packages/ui/src/Radio.module.scss

```diff
@@ -0,0 +1,109 @@
+@use '../styles/constants' as c;
+@use '../styles/helpers' as h;
+
+.wrapper {
+  display: inline-flex;
+  // inline-flex changes vertical align compared to display: flex
+  // setting vertical-align allows us to easily use radio button
+  // inline e.g. next to the text
+  // Most common use case will be still the usage in a form where parent
+  // wrapper controls this.
+  vertical-align: bottom;
+  align-items: center;
+  cursor: pointer;
+
+  @include c.typographyBodySmall;
+  // we need to provide spacing between radios if they're in a vertical column
+  margin-bottom: c.$grid;
+
+  // hide native checkbox
+  input {
+    @include h.visuallyHidden;
+  }
+
+  // default visible checkmark
+  .checkmark {
+    width: c.$iconSizeSmall;
+    height: c.$iconSizeSmall;
+    border: c.$borderWidth solid c.$colorNeutralLight;
+    border-radius: 50%;
+    // because of ~ and + selectors used below, we need to keep the current order in DOM and
+    // fix it visually with order property
+    order: -1;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &.inline {
+    .checkmark {
+      // in case we have inline mode, we should have a padding like native radio button has
+      margin-left: c.$grid;
+    }
+  }
```

Comment:
What if there is no leading text before the group of the radios? Wouldn't the margin feel weird then? I would suggest adding something like:
```
.radioGroup {
  display: flex;
  flex-direction: column;

  &.inline > *:not(:first-child) {
    margin-left: c.$grid;
  }
}
```
to the radio group instead. Then we would be able to remove the `inline` from the context.
I see why you wanted to add the `inline` to the context just like we did with the `error`. However, in my mind they are two different beasts. When we styled the error from the radio group we relied on the **implementation detail** of the radio. In other words, we had to know how it is implemented. And we affected that implementation from the outside.
With the margin, we are styling children's spacing as black boxes. We do not change their internal implementation. 
Does it make my sense? WDYT?

Reply:
I never liked passing it through the context, but just to make sure we realise this, this kills the ability of having intermediate DOM elements in between RadioGroup and Radio elements. But I'm fine with that.

Reply:
@jvorcak How exactly does this kill our Radio Inline scenario? What kind of other elements would you like to insert there?

Reply:
@jvorcak I feel like we do not have too many options here:
1. Use the `inline` prop in the context. Pro: Radio will be able to add margin on its own. Con: Radio will always have the margin, even it is the first Radio in the row. Additionally, it kind of breaks encapsulation see https://github.com/hazelcast/frontend-shared/pull/54#discussion_r520589718
2. Let the parent handle the margin. Agree that it will work only if Radios are direct descendants. However, the user can always override the class for more complex scenarios. And are there any complex scenarios in the first place?

Reply:
Yeah, I agree with number 2 and that is the current implementation, so you can resolve if you're happy with it.